homeassistant/components/hydrawise/sensor.py

```diff
@@ -63,15 +119,51 @@ class HydrawiseSensor(HydrawiseEntity, SensorEntity):
 
     def _update_attrs(self) -> None:
         """Update state attributes."""
-        if self.entity_description.key == "watering_time":
-            if (current_run := self.zone.scheduled_runs.current_run) is not None:
-                self._attr_native_value = int(
-                    current_run.remaining_time.total_seconds() / 60
-                )
-            else:
-                self._attr_native_value = 0
-        elif self.entity_description.key == "next_cycle":
-            if (next_run := self.zone.scheduled_runs.next_run) is not None:
-                self._attr_native_value = dt_util.as_utc(next_run.start_time)
-            else:
-                self._attr_native_value = datetime.max.replace(tzinfo=dt_util.UTC)
+        specific_update_attrs_method = getattr(
+            self, f"_update_attrs_{self.entity_description.key}"
+        )
+        if specific_update_attrs_method:
+            specific_update_attrs_method()
+
+    def _update_attrs_watering_time(self) -> None:
```

Comment:
I'm not a fan of all the side effects being introduced by these. I'd rather them just return values and have `_update_attrs()` assign the value to `self._attr_native_value`.

Reply:
Fixed.